### PR TITLE
propagate getNodeFromDb err

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- If somehow a trie node is missing, the returned error is masked.
  
## Proposed Changes
- Propagate the getNodeFromDB error until baseSync. This will automatically trigger the state sync.

## Testing procedure
- Normal testing with missingTrieNodes sed

### CODE REVIEW

File: process/smartContract/process.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go-core/marshal"
 	logger "github.com/ElrondNetwork/elrond-go-logger"
 	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/errors"
```
Comment (iulianpascalau) [inline process/smartContract/process.go:22]: reminder for self: do not forget about the smart contract processor v2 update

**Review by iulianpascalau — COMMENTED** [review]
looks good :+1:

File: process/smartContract/process.go [hunk 2]
```diff
@@ -342,6 +343,9 @@ func (sc *scProcessor) doExecuteSmartContractTransaction(
 	var results []data.TransactionHandler
 	results, err = sc.processVMOutput(vmOutput, txHash, tx, vmInput.CallType, vmInput.GasProvided)
 	if err != nil {
+		if errors.IsGetNodeFromDBError(err) {
+			return vmcommon.UserError, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:347]: I do not think there is a need for a new error. Changed to `vmcommon.ExecutionFailed`

File: process/smartContract/process.go [hunk 3]
```diff
@@ -377,6 +381,9 @@ func (sc *scProcessor) executeSmartContractCall(
 	vmOutput, err = vmExec.RunSmartContractCall(vmInput)
 	sc.arwenChangeLocker.RUnlock()
 	if err != nil {
+		if errors.IsGetNodeFromDBError(err) {
+			return userErrorVmOutput, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:385]: I do not think we do. Changed to return `nil, err`

File: process/smartContract/process.go [hunk 4]
```diff
@@ -937,6 +944,9 @@ func (sc *scProcessor) doExecuteBuiltInFunction(
 		tmpCreatedAsyncCallback := false
 		tmpCreatedAsyncCallback, newSCRTxs, err = sc.processSCOutputAccounts(newVMOutput, vmInput.CallType, outPutAccounts, tx, txHash)
 		if err != nil {
+			if errors.IsGetNodeFromDBError(err) {
+				return vmcommon.ExecutionFailed, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:948]: `vmcommon.ExecutionFailed` should be ok

File: process/smartContract/process.go [hunk 5]
```diff
@@ -1036,6 +1046,10 @@ func (sc *scProcessor) resolveBuiltInFunctions(
 			GasRemaining:  0,
 		}
 
+		if errors.IsGetNodeFromDBError(err) {
+			return vmOutput, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:1050]: Done.

File: process/smartContract/process.go [hunk 6]
```diff
@@ -1664,6 +1679,9 @@ func (sc *scProcessor) doDeploySmartContract(
 	sc.arwenChangeLocker.RUnlock()
 	if err != nil {
 		log.Debug("VM error", "error", err.Error())
+		if errors.IsGetNodeFromDBError(err) {
+			return vmcommon.UserError, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:1683]: Changed to `return vmcommon.ExecutionFailed, err`

File: process/smartContract/process.go [hunk 7]
```diff
@@ -1686,6 +1704,9 @@ func (sc *scProcessor) doDeploySmartContract(
 	results, err := sc.processVMOutput(vmOutput, txHash, tx, vmInput.CallType, vmInput.GasProvided)
 	if err != nil {
 		log.Trace("Processing error", "error", err.Error())
+		if errors.IsGetNodeFromDBError(err) {
+			return vmcommon.ExecutionFailed, err
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/smartContract/process.go:1708]: I believe not.

File: process/transaction/shardProcess.go [hunk 8]
```diff
@@ -217,6 +218,10 @@ func (txProc *txProcessor) executeAfterFailedMoveBalanceTransaction(
 	tx *transaction.Transaction,
 	txError error,
 ) error {
+	if elrondErr.IsGetNodeFromDBError(txError) {
+		return txError
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/transaction/shardProcess.go:222]: I believe not.

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**